resty/example/transformer.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+import 'package:http/http.dart' as ht;
+import 'package:jaguar_resty/jaguar_resty.dart' as resty;
+import 'package:jaguar/jaguar.dart';
+
+// TODO WIP
+
+final server = Jaguar(port: 8000)
+  ..getJson('/math/addition',
+      (Context ctx) => ctx.query.getInt("a") + ctx.query.getInt("b"));
+
+resty.Before setQuery(int a, int b) => (resty.RouteBase route) {
+      route.query("a", a);
+      route.query("b", b);
+    };
+
+void printResult(resty.Response<String> resp) {
+  print(resp.body);
+}
+
+Future client() async {
+  await resty
+      .get('/math/addition')
+      .http('localhost:8000')
+      .interceptBefore(setQuery(5, 20))
```

Comment:
so here it can be Interceptor class object or Before ?

Reply:
It can be both. `Interceptor` class implements `Before` method through Dart's `call` semantics.

Reply:
Ok :)